### ISSUE TITLE
About automatic IP address assignment to UE's TUN interface

### ISSUE THREAD

**Issue by s5uishida** [aligungr/UERANSIM#195]
Hi @aligungr 

It would be very helpful if the IP address assigned to the UE's TUN interface from the core network is automatically set to the UE's TUN interface.

I'm sorry for many requests.

**Comment by aligungr** [comment]
Hi @s5uishida. This feature is now on our roadmap, thanks for the suggestion.

**Comment by s5uishida** [comment]
Thank you very much!

**Comment by aligungr** [comment]
Hi @s5uishida

With v2.2.1 we implemented some of your suggestions.

1. UERANSIM now supports multiple `uesimtunX` interface as you suggested in https://github.com/aligungr/UERANSIM/issues/192. A TUN interface is created for each PDU session.
2. TUN interface is automatically configured and IP address is automatically assigned to the TUN interface. Indeed we removed all manual configurations for the TUN interface.
3. And lastly, now we support multiple gNBs in the same machine, using different network interfaces.

However, your other suggestion - which is the complete seperation of UE and gNB - is still in roadmap, but we're planning to complete soon. Currently UE and gNB must be in the same machine.

Thanks for the suggestions.

**Comment by s5uishida** [comment]
Hi @aligungr 

It's excellent!

Here, I created 5 TUN interfaces on one VM as shown below, and confirmed simultaneous ping via each specified U-Planes.

- uesimtun0 : 10.45.0.2 (dnn=internet)
- uesimtun1 : 10.46.0.2 (dnn=internet2)
- uesimtun2 : 10.46.0.3 (dnn=internet2)
- uesimtun3 : 10.47.0.2 (dnn=ims)
- uesimtun4 : 10.47.0.3 (dnn=ims)

I haven't tried using multiple gNB NICs at the same time yet, but I was able to connect many UEs to different DNNs with one VM. Very helpful!

I would like to update my procedure memo half a day later,

Thank you very much again!!
